### ISSUE TITLE
pandas version constraint not compatible with python 3.7

### ISSUE THREAD

**Issue by dbwpe** [DeepSpace2/StyleFrame#52]
Hey there, I happened across [this stackoverflow](https://stackoverflow.com/questions/54227936/i-encounter-this-error-when-i-try-pip-install-styleframe-any-idea-what-this-e) today while perusing Newest, figured I'd report here what I found while trying to assist.

Attempting to install with python 3.7, the pandas version constraint (`pandas>=0.16.2,<=0.23.*`) results in `pandas==0.22.0`, which results in `numpy==1.13.1`, which is not compatible with python 3.7 (it fails trying to build).  pandas 0.23.4 (latest) installs clean.

I dunno know how much work it would be to make styleframe work with latest pandas (I see a couple closed issues that make it seem easy?), but methinks that would solve the issue.

If you have docker, you can replicate with `docker run --rm -it python:3.7 pip install styleframe`

As a sidenote, while setup.py here includes python 3.7, .travis.yml does not.  Presumably travis would catch this issue if tests included 3.7.

Let me know if you'd like me to put up a PR to see if tests pass :)

**Comment by DeepSpace2** [comment]
Hi, thanks for detailed info + the answer on SO :)

`travis.yml` actually included Python 3.7 on branch `devel` but was removed in a later commit. The result can be seen here (https://travis-ci.org/DeepSpace2/StyleFrame/jobs/474158775). Indeed `pandas>=0.16.2,<=0.23.*` results with `pandas==0.22.0` with Python 3.7 (not sure why though, I guess `pandas>0.22` itself does not "support" Python 3.7) which in turn caused the `numpy` build to fail.

I'm not sure how to solve it. Maybe we can try to increase the minimal `numpy` version?

**Comment by dbwpe** [comment]
Sure thing!

If I install pandas directly with no version constraint, I get:

```
docker run --rm -it python:3.7 pip install pandas
.........
Successfully installed numpy-1.16.0 pandas-0.23.4 python-dateutil-2.7.5 pytz-2018.9 six-1.12.0
```

After I got some sleep, it looks like 0.23.2 is the first version of pandas that works with 3.7:
https://pandas.pydata.org/pandas-docs/version/0.23.2/whatsnew.html

Can confirm in docker that 0.23.2 installs but 0.23.1 does not.

So if you can finagle this codebase up to at least pandas 0.23.2 (ideally up to latest 0.23.4 to kick the can further down the road), then that should make happy.

**Comment by DeepSpace2** [comment]
I opted for `pandas==0.23.4` (any newer version will not support Python 2.7) but for some odd reason it is not available for Python 3.4 🤔 (https://travis-ci.org/DeepSpace2/StyleFrame/jobs/481125491#L196), works for any other version though (https://travis-ci.org/DeepSpace2/StyleFrame/builds/481125489)

Any idea why  `pandas>=0.16.2,<=0.23.*` insists on pulling 0.22.0?

**Comment by DeepSpace2** [comment]
Problem solved :) (latest travis build https://travis-ci.org/DeepSpace2/StyleFrame/builds/481319880).

Turns out PEP-508 introduced 'environment markers' that allow conditional versioning in `requierments.txt` and `setup.py`.

https://github.com/DeepSpace2/StyleFrame/blob/b7d58836e6dfe881aad8d63176b38bd9e75b6ec1/requirements.txt#L1-L2

https://github.com/DeepSpace2/StyleFrame/blob/b7d58836e6dfe881aad8d63176b38bd9e75b6ec1/setup.py#L66-L74
